src/app/combat-state.service.ts

```diff
@@ -1,19 +1,25 @@
 import { Injectable } from '@angular/core';
-import { Battlefield } from './classes/Battlefield';
 import { Game } from './Game';
+import { Battlefield } from './classes/Battlefield';
 import { Targetable } from './interfaces/Targetable';
+import { CombatState } from './state-trackers/CombatState';
+
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: 'root'
 })
-export class BattlefieldService {
+export class CombatStateService {
```

Comment:
this class literally doesn't do anything except pass through to another module's public functions. you don't need this. everything that calls combatstateservice can just call game directly. you don't need to be dependency injecting objects that don't do anything or hold any state themselves.

Reply:
you do the same thing in inventoryservice as well. it doesn't need to exist; in general using angular services is unnecessary unless you need to use dependency-injected angular stuff like http services. most of the time you can just write ordinary modules or classes and skip all the extra allocation, dependency injection, and boilerplate.

Reply:
Services have been removed. Thank you for pointing this out. As you know, I'm not super familiar with Angular, so I appreciate the guidance on Proper Angular Methods.